src/ui/marker.js

```diff
@@ -11,27 +11,43 @@ import type Popup from './popup';
 import type {LngLatLike} from "../geo/lng_lat";
 import type {MapMouseEvent} from './events';
 
+export type Anchor = 'middle';
+export type Offset = number | PointLike | {[Anchor]: PointLike};
+
+export type MarkerOptions = {
+    anchor: Anchor,
+    offset: Offset
+}
+
 /**
  * Creates a marker component
- * @param element DOM element to use as a marker. If left unspecified a default SVG will be created as the DOM element to use.
- * @param options
- * @param options.offset The offset in pixels as a {@link PointLike} object to apply relative to the element's center. Negatives indicate left and up.
+ * @param {Object} [element] DOM element to use as a marker. If left unspecified a default SVG will be created as the DOM element to use.
+ * @param {Object} [options]
+ * @param {string} [options.anchor] - A string indicating the markers's location relative to
+ *   the coordinate set via {@link Marker#setLngLat}.
+ *   Options is  `'middle'` by default
```

Comment:
See https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/map.js#L171 for an example of how to set the default value directly in the JSDoc. The documentation generator automatically picks up this default value as you can see https://www.mapbox.com/mapbox-gl-js/api/#map

Reply:
Also like https://github.com/mapbox/mapbox-gl-js/blob/d11cf304f8cb4554fdf46a95bd20eeb3df090c59/src/ui/map.js#L171 I'd include a list of all the valid options for 'anchor'

Reply:
omg, if this works i'm probably going to cry lol 😂